internal/tiltfile/starkit/set_state.go

```diff
@@ -0,0 +1,62 @@
+package starkit
+
+import (
+	"fmt"
+	"reflect"
+
+	"go.starlark.net/starlark"
+)
+
+// SetState works like SetState in React. It can take a value or a function.
+//
+// That function should transform old state into new state.
+//
+// For example, an extension that accumulated strings might use
+// SetState() like this:
+//
+// err := starkit.SetState(t, func(strings []string) {
+//   return append([]string{newString}, strings...)
+// })
+//
+// This would be so much easier with generics :grimace:
+//
+// SetState will return an error if it can't match the type
+// of anything in the state store.
+func SetState(t *starlark.Thread, valOrFn interface{}) error {
+	typ := reflect.TypeOf(valOrFn)
+	model, ok := t.Local(modelKey).(Model)
+	if !ok {
+		return fmt.Errorf("Internal error: Starlark not initialized correctly: starkit.Model not found")
+	}
+
+	isFn := typ.Kind() == reflect.Func
+	if !isFn {
+		// If there's already a value with this type in the state store, overwrite it.
```

Comment:
hmm, does this mean we can't have a state that stores, e.g., two different `int` values?

Reply:
ya, i would expect that every extension will have its own type of state